sway-lib-core/src/ops.sw

```diff
@@ -472,3 +511,38 @@ impl OrdEq for u16 {
 }
 impl OrdEq for u8 {
 }
+
+/////////////////////////////////////////////////
+// Internal Helpers
+/////////////////////////////////////////////////
+
+/// Extract a single 64 bit word from a b256 value using the specified offset.
+fn get_word_from_b256(val: b256, offset: u64) -> u64 {
+    let mut empty: u64 = 0;
+    asm(r1: val, offset: offset, r2,  res: empty) {
+        add r2 r1 offset;
+        lw res r2 i0;
+        res: u64
```

Comment:
It doesn't matter now that it's merged, but I feel like there's a lack of symmetry here.  We know there's 4 words in a `b256` so the index/offset might as well be word-wise and then the `ADD` isn't needed, and it kinda works in reverse to `compose()`.  Seeing the `get_word_from_b256(8)` below felt a bit... wrong. 🙂

Reply:
@otrho I feel it's worth striving for the best & cleanest code we can produce, so I'm happy to push a change to this when I have a moment, won't take long. 🙂